test/persistence-hooks.suite.js

```diff
@@ -234,6 +234,45 @@ module.exports = function(dataSource, should) {
           });
       });
 
+      it('triggers `loaded` hook', function(done) {
+        TestModel.observe('loaded', pushContextAndNext());
+
+        TestModel.create(
+          { id: 'new-id', name: 'a name' },
+          function(err, instance) {
+            if (err) return done(err);
+
+            observedContexts.should.eql(aTestModelCtx({
+              data: { id: 'new-id', name: 'a name' },
+              isNewInstance: true
+            }));
+
+            done();
+          });
+      });
+
+      // TODO (pradnya) - the, instance returned by `create` context 
+      // does not have values updated from `loaded` hook. 
+      // Gives error - Cannot read property 'should' of undefined 
```

Comment:
There is no `instance.data` AFAIK, I think you should assert directly on the `instance` object. Alternatively you can call `instance.toObject()` and assert on that.